packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -1250,22 +1250,26 @@ abstract class ResidentRunner {
     }
   }
 
-  Future<bool> launchDevTools({bool openInBrowser = false}) async {
+  DevToolsServerAddress activeDevToolsServer() {
+    _devToolsLauncher ??= DevtoolsLauncher.instance;
+    return _devToolsLauncher.activeDevToolsServer;
+  }
+
+  Future<void> serveDevToolsGracefully({bool openInBrowser = false}) async {
     if (!supportsServiceProtocol) {
-      return false;
+      return;
+    }
+    try {
+      _devToolsLauncher ??= DevtoolsLauncher.instance;
+      return await _devToolsLauncher.serve(openInBrowser: openInBrowser);
```

Comment:
@jonahwilliams I think this being awaited is causing tests to fail with timeouts. However, trying to leave this unawaited also caused issues in the tests `Bad state: A Timer was active at the end of a test`. Should we be awaiting here?

Reply:
I think you'll need to make sure that any tests that call this are injecting a faked Devtools launcher instead

Reply:
done